analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant.cs

```diff
@@ -63,16 +70,131 @@ public sealed class MemberInitializerRedundant : SonarDiagnosticAnalyzer
                     // interfaces cannot have instance fields and instance properties cannot have initializers
                     if (declaration is ClassDeclarationSyntax)
                     {
-                        InitializationVerifierFactory.InstanceMemberInitializationVerifier.Report(c, declaration, members);
+                        InitializationVerifierFactory.InstanceMemberInitializationVerifier.Report(c, declaration, members, forceSonarCfg);
                     }
-                    InitializationVerifierFactory.StaticMemberInitializationVerifier.Report(c, declaration, members);
+                    InitializationVerifierFactory.StaticMemberInitializationVerifier.Report(c, declaration, members, forceSonarCfg);
                 },
                 // For record support, see details in https://github.com/SonarSource/sonar-dotnet/pull/4756
                 // it is difficult to work with instance record constructors w/o raising FPs
                 SyntaxKind.ClassDeclaration,
                 SyntaxKind.StructDeclaration,
                 SyntaxKind.InterfaceDeclaration);
 
+        private class RedundancyChecker
+        {
+            private readonly ISymbol memberToCheck;
+            private readonly SemanticModel semanticModel;
+
+            public RedundancyChecker(ISymbol memberToCheck, SemanticModel semanticModel)
+            {
+                this.memberToCheck = memberToCheck;
+                this.semanticModel = semanticModel;
+            }
+
+            public bool IsMemberUsedInsideLambda(SyntaxNode instruction) =>
+                instruction.DescendantNodes()
+                    .OfType<IdentifierNameSyntax>()
+                    .Select(GetPossibleMemberAccessParent)
+                    .Any(IsMatchingMember);
+
+            public static ExpressionSyntax GetPossibleMemberAccessParent(SyntaxNode node)
```

Comment:
Same for the other one
```suggestion
            public static ExpressionSyntax PossibleMemberAccessParent(SyntaxNode node)
```